src/proof/PGTransformationAlgorithms.cc

```diff
@@ -208,96 +209,74 @@ double ProofGraph::recyclePivotsIter() {
+                    mergeClauses(n->getAnt1()->getClause(), n->getAnt2()->getClause(), n->getClause(), n->getPivot());
+                    for (clauseid_t clauseid : n->getResolvents()) {
+                        if (getNode(clauseid)) { q.push_back(clauseid); }
+                    }
                     // NOTE extra check
-                    if (proofCheck() > 1) checkClause(replacing->getId());
+                    if (proofCheck() > 1) checkClause(n->getId());
                 } else {
-                    //Look for pivot in antecedents
-                    bool piv_in_ant1 = false, piv_in_ant2 = false;
-                    short f1 = n->getAnt1()->hasOccurrenceBin(n->getPivot());
-                    if (f1 != -1) { piv_in_ant1 = true; }
-                    short f2 = n->getAnt2()->hasOccurrenceBin(n->getPivot());
-                    if (f2 != -1) { piv_in_ant2 = true; }
-                    assert(not(f1 == 1 and f2 == 1) and not(f1 == 0 and f2 == 0));
-
-                    if (piv_in_ant1 and piv_in_ant2) {
-                        //Easy case: pivot still in both antecedents
-                        //Sufficient to propagate modifications via merge
-                        mergeClauses(n->getAnt1()->getClause(), n->getAnt2()->getClause(), n->getClause(), n->getPivot());
-                        for (clauseid_t clauseid : n->getResolvents()) {
-                            if (getNode(clauseid)) { q.push_back(clauseid); }
-                        }
-                        // NOTE extra check
-                        if (proofCheck() > 1) checkClause(n->getId());
+                    assert(not piv_in_ant1 or not piv_in_ant2);
+                    //Second case: pivot not in ant1 or not in ant2
+                    //Remove resolution step, remove n, ant without pivots gains n resolvents
+                    bool choose_ant1 = false;
+                    //Pivot not in ant1 and not in ant2
+                    if (not piv_in_ant1 and not piv_in_ant2) {
+                        // Choose one of the two antecedents heuristically
+                        choose_ant1 = chooseReplacingAntecedent(n);
                     } else {
-                        assert(not piv_in_ant1 or not piv_in_ant2);
-                        //Second case: pivot not in ant1 or not in ant2
-                        //Remove resolution step, remove n, ant without pivots gains n resolvents
-                        bool choose_ant1 = false;
-                        //Pivot not in ant1 and not in ant2
-                        if (not piv_in_ant1 and not piv_in_ant2) {
-                            // Choose one of the two antecedents heuristically
-                            choose_ant1 = chooseReplacingAntecedent(n);
-                        } else {
-                            choose_ant1 = not piv_in_ant1 and piv_in_ant2;
-                        }
+                        choose_ant1 = not piv_in_ant1 and piv_in_ant2;
```

Comment:
Here we know from the assert in line 230 and the condition in line 235 that either `pid_in_ant1 and not piv_in_ant2`, or `not piv_in_ant1 and piv_in_ant2`.  Maybe it would be more natural to say simply
```
choose_ant1 = not piv_in_ant1;
```
since the latter is implied.  Meaning that we `choose_ant1` if and only if the pivot is not there.

Reply:
Yeah I was considering it, then decided to not change the code, but I agree with you. I'll update it now.

Reply:
Fixed.